src/execute.c

```diff
@@ -917,11 +917,11 @@ jv jq_next(jq_state *jq) {
       }
       struct cfunction* function = &frame_current(jq)->bc->globals->cfunctions[*pc++];
       switch (function->nargs) {
-      case 1: top = ((jv (*)(jq_state *, jv))function->fptr)(jq, in[0]); break;
-      case 2: top = ((jv (*)(jq_state *, jv, jv))function->fptr)(jq, in[0], in[1]); break;
-      case 3: top = ((jv (*)(jq_state *, jv, jv, jv))function->fptr)(jq, in[0], in[1], in[2]); break;
-      case 4: top = ((jv (*)(jq_state *, jv, jv, jv, jv))function->fptr)(jq, in[0], in[1], in[2], in[3]); break;
-      case 5: top = ((jv (*)(jq_state *, jv, jv, jv, jv, jv))function->fptr)(jq, in[0], in[1], in[2], in[3], in[4]); break;
+      case 1: top = function->fptr.a1(jq, in[0]); break;
+      case 2: top = function->fptr.a2(jq, in[0], in[1]); break;
+      case 3: top = function->fptr.a3(jq, in[0], in[1], in[2]); break;
+      case 4: top = function->fptr.a4(jq, in[0], in[1], in[2], in[3]); break;
+      case 5: top = function->fptr.a5(jq, in[0], in[1], in[2], in[3], in[4]); break;
       // FIXME: a) up to 7 arguments (input + 6), b) should assert
       // because the compiler should not generate this error.
```

Comment:
Unrelated to this PR: is the numbers in this comment outdated? should say "up to 5 (input + 4)"? and is `MAX_CFUNCTION_ARGS` constant a bit weirdly named? is more a constant that is at least equal or more than what max c-builtins arity is?

Reply:
I also noticed the `MAX_CFUNCTION_ARGS` set to 10, but I have not checked what it does

Reply:
I set `MAX_CFUNCTION_ARGS` to 4, removed `cfunction.fptr.a5` since it was not used, and removed a bunch of unnecessary roundabout code from CALL_BUILTIN

Reply:
Nice 👍